monkey/infection_monkey/exploit/zerologon.py

```diff
@@ -0,0 +1,433 @@
+"""
+Zerologon, CVE-2020-1472
+Implementation based on https://github.com/dirkjanm/CVE-2020-1472/ and https://github.com/risksense/zerologon/.
+"""
+
+import logging
+import os
+import re
+from binascii import unhexlify
+from typing import Dict, List, Optional
+
+import impacket
+from impacket.dcerpc.v5 import nrpc
+from impacket.dcerpc.v5.dtypes import NULL
+
+from common.utils.exploit_enum import ExploitType
+from infection_monkey.exploit.HostExploiter import HostExploiter
+from infection_monkey.exploit.zerologon_utils.dump_secrets import DumpSecrets
+from infection_monkey.exploit.zerologon_utils.options import \
+    OptionsForSecretsdump
+from infection_monkey.exploit.zerologon_utils.wmiexec import Wmiexec
+from infection_monkey.network.zerologon_fingerprint import ZerologonFinger
+from infection_monkey.utils.capture_output import StdoutCapture
+
+LOG = logging.getLogger(__name__)
+
+
+class ZerologonExploiter(HostExploiter):
+    _TARGET_OS_TYPE = ['windows']
+    _EXPLOITED_SERVICE = 'Netlogon'
+    EXPLOIT_TYPE = ExploitType.VULNERABILITY
+    RUNS_AGENT_ON_SUCCESS = False
+    MAX_ATTEMPTS = 2000
+    ERROR_CODE_ACCESS_DENIED = 0xc0000022
+
+    def __init__(self, host: object):
+        super().__init__(host)
+        self.vulnerable_port = None
+        self.zerologon_finger = ZerologonFinger()
+        self.exploit_info['credentials'] = {}
+
+    def _exploit_host(self) -> bool:
+        self.dc_ip, self.dc_name, self.dc_handle = self.zerologon_finger._get_dc_details(self.host)
```

Comment:
The same problem: exploiter re-runs fingerprinter's code. Why can't fingerprinter add this info to the host? And here, we'd only check if it's added?

Reply:
Removing the fingerprinter as per yesterday's discussion.